tests/util/golinter/rules/lint_rules.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+// LintRule is interface for defining lint rules.
+type LintRule interface {
+	// GetID returns ID of the rule in string, ID is equal to the file name of that rule.
+	GetID() string
+	// Check verifies if aNode passes rule check. If verification fails it adds report into rpt.
+	Check(aNode ast.Node, fs *token.FileSet, rpt *[]string)
```

Comment:
Perhaps Check can take a Report object instead of a pointer to a string array, so Report can encapsulate the related util methods, and possibly make the report more structural for other consumption later.

Reply:
Done. Thanks!